File: backend/Services/ClientService.cs
using backend.Data;
using backend.Models;
using Microsoft.EntityFrameworkCore;

namespace backend.Services;

public class ClientService
{
    private readonly AppDbContext _db;

    public ClientService(AppDbContext db)
    {
        _db = db;
    }

    public async Task<List<Client>> GetClients(string username)
    {
        return await _db.Clients
            .Where(c => c.User.Username == username)
            .ToListAsync();
    }

    public async Task<Client> CreateClient(Client client, string username)
    {
        var user = await _db.Users.FirstAsync(u => u.Username == username);
        client.UserId = user.Id;
        _db.Clients.Add(client);
        await _db.SaveChangesAsync();
        return client;
    }

    public async Task<Client> UpdateClient(int id, Client updated, string username)
    {
        var client = await _db.Clients
            .FirstOrDefaultAsync(c => c.Id == id && c.User.Username == username)
            ?? throw new Exception("Client not found");

        client.Name = updated.Name;
        client.Afm = updated.Afm;
        client.Address = updated.Address;
        client.City = updated.City;
        client.Country = updated.Country;

        await _db.SaveChangesAsync();
        return client;
    }

    public async Task DeleteClient(int id, string username)
    {
        var client = await _db.Clients
            .FirstOrDefaultAsync(c => c.Id == id && c.User.Username == username)
            ?? throw new Exception("Client not found");

        _db.Clients.Remove(client);
        await _db.SaveChangesAsync();
    }
}

File: frontend/src/app/services/client.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

export interface Client {
  id?: number;
  name: string;
  afm: string;
  address?: string;
  city?: string;
  country: string;
}

@Injectable({ providedIn: 'root' })
export class ClientService {
  private baseUrl = '/api/clients';

  constructor(private http: HttpClient) {}

  getClients() {
    return this.http.get<Client[]>(this.baseUrl);
  }

  createClient(client: Client) {
    return this.http.post<Client>(this.baseUrl, client);
  }

  updateClient(id: number, client: Client) {
    return this.http.put<Client>(`${this.baseUrl}/${id}`, client);
  }

  deleteClient(id: number) {
    return this.http.delete(`${this.baseUrl}/${id}`);
  }
}
